ozaria/site/components/teacher-dashboard/modals/ModalDynamicPromotion.vue

```diff
@@ -0,0 +1,98 @@
+<script>
+import Modal from '../../common/Modal'
+import utils from 'core/utils'
+import trackable from 'app/components/mixins/trackable.js'
+
+export default Vue.extend({
+  components: {
+    Modal,
+  },
+  mixins: [trackable],
+  props: {
+    title: {
+      type: String,
+      default: null,
+      required: false
+    },
+    seenPromotionsProperty: {
```

Comment:
this is so that you want modal to work even if we don't use the seenPromotion property to save and get?

Reply:
yes, the plan was something like that. also different modals should use different keys to differentiate what modals were seen.